src/Geolocator.Plugin.Android.Fused/FusedGeolocatorImplementation.cs

```diff
@@ -0,0 +1,301 @@
+						{
+							Longitude = address.Longitude,
+							Latitude = address.Latitude,
+							FeatureName = address.FeatureName,
+							PostalCode = address.PostalCode,
+							SubLocality = address.SubLocality,
+							CountryCode = address.CountryCode,
+							CountryName = address.CountryName,
+							Thoroughfare = address.Thoroughfare,
+							SubThoroughfare = address.SubThoroughfare,
+							Locality = address.Locality
+						});
+            }
+        }
+
+        public async Task<bool> StartListeningAsync(TimeSpan minimumTime, double minimumDistance, bool includeHeading = false, ListenerSettings listenerSettings = null)
+        {
+			var hasPermission = await CheckPermissions();
+			if (!hasPermission)
+				throw new GeolocationException(GeolocationError.Unauthorized);
+			
+            if (_client == null)
+			{
+				Initialize();
+			}
+
+            if (!_client.IsConnected)
+                _client.Connect();
+
+            if (!_client.IsConnected)
+                return await Task.FromResult(false);
+            
+            var minTime = (long)minimumTime.TotalMilliseconds;
+            var locationRequest = new LocationRequest();
+            locationRequest.SetSmallestDisplacement(Convert.ToSingle(minimumDistance))
+                .SetFastestInterval(minTime)
+                .SetInterval(minTime * 3)
+                .SetMaxWaitTime(minTime * 6)
+                .SetPriority(GetPriority());
+
+            var result = await LocationServices.FusedLocationApi.RequestLocationUpdatesAsync(_client, locationRequest, this);
+
+            if (result.IsSuccess)
+                IsListening = true;
+
+            return result.IsSuccess;
+        }
+
+        public async Task<bool> StopListeningAsync()
+        {
+			if (_client == null)
+				return true;
+
+            if (_client.IsConnected)
+            {
+                var result = await LocationServices.FusedLocationApi.RemoveLocationUpdatesAsync(_client, this);
+                if (!result.IsSuccess)
+                    return false;
+                
+				IsListening = false;
+            }
+
+			if (_client.IsConnected || _client.IsConnecting)
+			{
+				_client.Disconnect();
+				_client = null;
+			}
+
+            return true;
+        }
+
+        private int GetPriority()
+        {
+            if (DesiredAccuracy < 50)
+                return LocationRequest.PriorityHighAccuracy;
+
+            if (DesiredAccuracy < 100)
+                return LocationRequest.PriorityBalancedPowerAccuracy;
+
+            if (DesiredAccuracy < 200)
+                return LocationRequest.PriorityLowPower;
+
+            return LocationRequest.PriorityNoPower;
+        }
+
+        private Task<Position> NextLocationAsync()
+        {
+            var locationSource = new TaskCompletionSource<Position>();
+            EventHandler<PositionEventArgs> handler = null;
+
+            handler = (sender, args) =>
+            {
+                PositionChanged -= handler;
```

Comment:
It appears that multiple quick requests to resolve the position can lead in a hanging task here since only one of the callers gets a position through the callback here, and the others remain in a waiting state on the `locationSource.Task`  that you return.